### DISCUSSION TITLE
a question when layout wrap layout

### DISCUSSION THREAD

**Discussion by LawssssCat** [jekyll/jekyll#9065] [Q&A] [answered]
two layout 

```md
---
layout: page
attrs: ['attrs01']
---
this layout name is 'home.html'
{{ attrs }}
```
```md
---
attrs: ['attrs02']
---
this layout name is 'page.html'
```

in my imagination, `{{ attrs }}` is `['attrs01', 'attrs02']`. but in reality, `{{ attrs }}` is `['attrs01']`.

how can i get 'page.html''s attrs in `home.html`?

**Comment by ashmaroli** [comment]
Technically, just `{{ attrs }}` should return an empty string because the data enclosed between the set of triple dashes is scoped to the current *`page`* only.

i.e. the correct notation would be `{{ page.attrs }}`. Any other behavior is an anomaly and should be reported as a bug.

Regarding the topic of this ticket, the reference object is `layout`.

i.e. For a file `foo.md` to access front matter variable, say `alpha`, off its layout `page` would need the construct: `{{ layout.alpha }}`. Similarly, for the layout `page` to access a variable `beta` from its layout, `base` would need the construct ` {{ layout.beta }}`.

Projecting downwards, therefore for the page `foo.md` to access variable `beta` from layout `base`, the needed construct would be `{{ page.layout.layout.beta }}`.

**Reply by LawssssCat** [reply]
I just tried printing in the browser one by one 

+ `{{page}} ` is  `{"layout"=>"home", ...}`. it  seems to be a object
+ `{{layout}}` is ` {"layout"=>"page", "attrs "=>["attrs01"]}`.  In this case, i can get `attrs01`.
+ But  the trouble is `{{page.layout}}` is `home`. it  seems to be a string **because `{{page.layout.layout}}` or `{{page.layout.attrs}}` is empty.**
+ and `{{layout.layout}}` is `page`. it alse  seems to be a string **because `{{layout.layout.attrs }}` is empty.** 

In the end,  i still can't get the ‘attrs02’  defined in page. It seems because that I can't get the parent class layout of layout.

Is there any solution? Or did I ignore something?

**Reply by ashmaroli** [reply]
You're right.
I apologise for misinforming.

~~Layout data doesn't trickle down infinitely.~~

~~So, what you're trying to do is not possible.~~

~~Its not a bug, just a lack of feature.~~

**Reply by LawssssCat** [reply]
Is there any alternative solution to get the data that different levels of layout are willing to give and iterate trough them on the page?

**Comment by ashmaroli** [comment, marked as answer]
@LawssssCat I dug into the codebase. This is what I found:

Layout data gets merged hierarchically (parent **onto** child). The resulting data is exposed to the *page* as the top-level object `{{ layout }}`.

So in the case of :

```
---
layout: home
---

# Home Page

<pre>
  {{ layout }}
</pre>
```
```
---
layout: page
foo   : alpha
lorem : ipsum
cats  : [one, two]
---

<!-- from 'home' layout -->

{{ content }}
```
```
---
foo  : beta
gamma: delta
cats: [three, four]
---

<!-- from 'page' layout -->

{{ content }}
```
will render the following:
```
{"layout"=>"page", "foo"=>"beta", "lorem"=>"ipsum", "cats"=>["three", "four"], "gamma"=>"delta"}
```

**Reply by LawssssCat** [reply]
thanks, this inspired me. 
i thought of the following method to iterate over the information from different layouts:
(although it seems a bit strange.)

```yml
# the file './_data/layout.html'
name:
  - page
  - home 
  - other
```
```html
---
layout: home
---

<h1 id="home-page">Home Page</h1>

<pre>{{layout}}</pre>

<pre>
{%- for layout_name in site.data.layout.name -%}
{{layout[layout_name]}}
{% endfor %}
</pre>
```
```html
---
layout: page
home: ['home info']
---

<!-- from 'home' layout -->

{{ content }}
```
```html
---
layout: page
other: ['other info']
---

<!-- from 'other' layout -->

{{ content }}
```
```html
---
page: ["page info"]
---

<!-- from 'page' layout -->

{{ content }}
```

the following renders:

```
{"layout"=>"page", "home"=>["home"], "page"=>["page"]}
page
home
```

**Reply by MuhittinTekin** [reply]
{"layout"=>"page", "foo"=>"beta", "lorem"=>"ipsum", "cats"=>["three", "four"], "gamma"=>"delta"}
